packages/@aws-cdk/core/test/test.arn.ts

```diff
@@ -97,6 +97,14 @@ export = {
     test.done();
   },
 
+  'fails if missing a component and Stack is undefined'(test: Test) {
+    test.throws(() => Arn.format({
+      service: 'foo',
+      resource: 'bar',
+    }));
```

Comment:
```suggestion
    }), /Stack must be defined if any individual component is undefined/);
```

or whatever you end up changing the error message to